### PR TITLE
domain: concurrently fetch schema tables.

### PR DESCRIPTION
**Pull request by coocood**
Reduce the chance of time out when we have too many databases and tables.

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -78,28 +81,56 @@ func (do *Domain) loadInfoSchema(handle *infoschema.Handle, usedSchemaVersion in
...
+		}
+	}
+	return allSchemas, nil
+}
+
+const fetchSchemaConcurrency = 8
+
+func (do *Domain) splitForConcurrentFetch(schemas []*model.DBInfo) [][]*model.DBInfo {
+	groupSize := (len(schemas) + fetchSchemaConcurrency - 1) / fetchSchemaConcurrency
+	var splitted [][]*model.DBInfo
```
Comment (ngaut) [inline domain/domain.go]: Make with length.

File: domain/domain.go [hunk 2]
```diff
@@ -78,28 +81,56 @@ func (do *Domain) loadInfoSchema(handle *infoschema.Handle, usedSchemaVersion in
...
 
-func (do *Domain) getAllSchemasWithTablesFromMeta(m *meta.Meta) ([]*model.DBInfo, error) {
-	schemas, err := m.ListDatabases()
+func (do *Domain) fetchAllSchemasWithTables(m *meta.Meta) ([]*model.DBInfo, error) {
+	allSchemas, err := m.ListDatabases()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	splittedSchemas := do.splitForConcurrentFetch(allSchemas)
+	doneChan := make(chan error, len(splittedSchemas))
```
Comment (ngaut) [inline domain/domain.go]: s/doneChan/doneCh/

File: domain/domain.go [hunk 3]
```diff
@@ -78,28 +81,56 @@ func (do *Domain) loadInfoSchema(handle *infoschema.Handle, usedSchemaVersion in
...
+			return nil, errors.Trace(err)
+		}
+	}
+	return allSchemas, nil
+}
+
+const fetchSchemaConcurrency = 8
+
+func (do *Domain) splitForConcurrentFetch(schemas []*model.DBInfo) [][]*model.DBInfo {
+	groupSize := (len(schemas) + fetchSchemaConcurrency - 1) / fetchSchemaConcurrency
```
Comment (ngaut) [inline domain/domain.go]: Add a new variable schemaCnt := len(schemas), to avoiding use len(schema) everywhere.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by coocood** [comment]
@ngaut PTAL

**Comment by ngaut** [comment]
LGTM. PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**